include/vcpkg/base/util.h

```diff
@@ -19,9 +19,15 @@ namespace vcpkg::Util
     namespace Vectors
     {
         template<class Container, class T>
-        void append(std::vector<T>* augend, const Container& addend)
+        void append(std::vector<T>* augend, Container&& addend)
         {
-            augend->insert(augend->end(), addend.begin(), addend.end());
+            augend->insert(
+                augend->end(), std::make_move_iterator(addend.begin()), std::make_move_iterator(addend.end()));
+        }
+        template<class Container, class T>
+        void append(std::vector<T>* augend, Container& addend)
+        {
+            augend->insert(augend->end(), addend.cbegin(), addend.cend());
         }
```

Comment:
This doesn't do the right thing either, because `Container&&` triggers perfect forwarding.

I think you need something like:

```c++
        template<class Container, class T>
        void append(std::vector<T>* augend, Container&& addend)
        {
            if constexpr (std::is_lvalue_reference_v<Container>) {
                augend->insert(augend->end(), addend.begin(), addend.end());
            } else {
                augend->insert(augend->end(), std::make_move_iterator(addend.begin()), std::make_move_iterator(addend.end()));
            }
        }
```

Reply:
Edit: my first try does the wrong thing with const:

```c++
        template<class Container, class T>
        void append(std::vector<T>* augend, Container&& addend)
        {
            if constexpr (std::is_lvalue_reference_v<Container> || std::is_const_v<Container>) {
                augend->insert(augend->end(), addend.begin(), addend.end());
            } else {
                augend->insert(augend->end(), std::make_move_iterator(addend.begin()), std::make_move_iterator(addend.end()));
            }
        }
```